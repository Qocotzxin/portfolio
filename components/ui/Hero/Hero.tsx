import { Flex, Heading, VStack } from "@chakra-ui/react";
import { useHygraphContext } from "@contexts/hygraphContext";
import { FC } from "react";
import Disk from "./Disk";

const headingProps = (right = true) => ({
  size: { base: "2xl", lg: "4xl" },
  bgGradient: `linear(to-${
    right ? "r" : "l"
  }, var(--chakra-colors-teal-600), var(--chakra-colors-yellow-400))`,
  bgClip: "text",
  _dark: {
    bgGradient: `linear(to-${
      right ? "r" : "l"
    }, var(--chakra-colors-teal-400), var(--chakra-colors-yellow-300))`,
  },
});

const Hero: FC = () => {
  const { heroes } = useHygraphContext();

  if (!heroes || !heroes[0]) {
    return null;
  }

  const { title, subtitle } = heroes[0];

  return (
    <Flex
      bg="gray.100"
      _dark={{ bg: "black" }}
      h={{ base: "100%", md: "75%" }}
      w="100%"
      color="white"
      overflow="hidden"
      className="fadeIn-slower"
      flexDir={{ base: "column", md: "row" }}
      alignItems="center"
      justifyContent="space-between"
      p="12"
      data-testid="Hero"
    >
      <VStack spacing={2} zIndex="1" className="fadeIn-slower">
        <Heading {...headingProps()} as="h1" pb="1">
          {title}
        </Heading>
        {subtitle && (
          <Heading {...headingProps(false)} as="h2">
            {subtitle}
          </Heading>
        )}
      </VStack>
      {/* Disk container to manipulate overflow */}
      <Disk alignSelf={{ base: "center", md: "baseline" }} />
    </Flex>
  );
};

export default Hero;
