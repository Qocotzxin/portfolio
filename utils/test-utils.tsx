import { ThemeProvider } from "@contexts/themeContext";
import { Components, HygraphData } from "@models/hygraph";
import { render, RenderOptions } from "@testing-library/react";
import { HygraphProvider } from "contexts/hygraphContext";
import { FC, PropsWithChildren, ReactElement } from "react";

export const hygraphMockData: Partial<HygraphData> = {
  skipLinks: [
    {
      text: "Skip to content",
    },
  ],
  ariaLabels: [
    {
      id: "1",
      metadata: "dark",
      component: Components.ColorModeSwitcher,
      content: "Switch to light mode",
    },
    {
      id: "2",
      metadata: "light",
      component: Components.ColorModeSwitcher,
      content: "Switch to dark mode",
    },
  ],
  navLinks: [
    {
      url: "test",
      order: 1,
      text: "Test",
    },
  ],
  languages: [
    { code: "en", displayName: "english", isActive: true },
    { code: "es", displayName: "spanish", isActive: true },
    { code: "nl", displayName: "dutch", isActive: false },
  ],
  footerLinks: [
    {
      url: "https://github.com",
      isActive: true,
      icon: "github",
      ariaLabel: "Visit Github",
    },
    {
      url: "https://test.com",
      isActive: false,
      icon: "github",
      ariaLabel: "Visit test",
    },
  ],
  heroes: [
    {
      title: "Hero title",
      subtitle: "Hero subtitle",
    },
  ],
};

interface AllTheProvidersProps {
  hygraphData?: Partial<HygraphData>;
}

export const AllTheProviders: FC<PropsWithChildren & AllTheProvidersProps> = ({
  children,
  hygraphData = hygraphMockData,
}) => {
  return (
    <ThemeProvider>
      <HygraphProvider hygraphData={hygraphData}>{children}</HygraphProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  wrapperProps?: AllTheProvidersProps
) =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllTheProviders {...wrapperProps}>{children}</AllTheProviders>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { customRender as render };
