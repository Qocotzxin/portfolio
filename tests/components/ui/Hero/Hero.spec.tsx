import Hero from "@scenes/Home";
import { render, screen } from "@utils/test-utils";

describe("Hero.", () => {
  it("Should not render the component when hero is not provided through Hygraph.", () => {
    render(<Hero />, {}, { hygraphData: { heroes: undefined } });

    expect(screen.queryByTestId("Hero")).not.toBeInTheDocument();
  });

  it("Should not render the component when heroes data is present but empty.", () => {
    render(<Hero />, {}, { hygraphData: { heroes: [] } });

    expect(screen.queryByTestId("Hero")).not.toBeInTheDocument();
  });

  it("Should render Hero and match snapshot when heroes ndata is available.", () => {
    const { container } = render(<Hero />);

    expect(screen.getByTestId("Hero")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("Should render Hero even when subtitle is not present.", () => {
    render(<Hero />);

    expect(screen.getByTestId("Hero")).toBeInTheDocument();
  });
});
