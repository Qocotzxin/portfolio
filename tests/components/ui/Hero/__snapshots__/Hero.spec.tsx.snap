// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero. Should render Hero and match snapshot when heroes ndata is available. 1`] = `
<div>
  <div
    class="Home"
    data-testid="Home"
  >
    <div
      class="Home Home__hero "
      data-testid="Hero"
    >
      <div
        class="Home__hero-title "
      >
        <h1>
          Hero title
        </h1>
        <h2>
          Hero subtitle
        </h2>
      </div>
      <svg
        class="Disk"
        height="100%"
        role="img"
        viewBox="0 0 500 500"
        width="100%"
      >
        <g
          class="Disk__innerContainer"
        >
          <ellipse
            cx="250"
            cy="250"
            fill="black"
            rx="200"
            ry="220"
            stroke="black"
            stroke-width="4"
          />
          <ellipse
            cx="250"
            cy="250"
            fill="#1a1a1a"
            rx="160"
            ry="180"
            stroke="#1a1a1a"
            stroke-width="4"
          />
          <ellipse
            cx="250"
            cy="250"
            fill="#ECD51B"
            rx="120"
            ry="150"
            stroke="#ECD51B"
            stroke-width="4"
          />
          <ellipse
            cx="250"
            cy="250"
            fill="#2449DA"
            rx="80"
            ry="100"
            stroke="#2449DA"
            stroke-width="4"
          />
          <ellipse
            cx="250"
            cy="250"
            fill="#D94214"
            rx="40"
            ry="60"
            stroke="#D94214"
            stroke-width="4"
          />
          <text
            x="227"
            y="258"
          >
            HTML
          </text>
          <text
            x="227"
            y="178"
          >
            CSS
          </text>
          <text
            x="234"
            y="142"
          >
            JS
          </text>
        </g>
      </svg>
    </div>
  </div>
</div>
`;
